### PR TITLE
feat: support spec as input

### LINKED ISSUE #396: Kusion supports spec as input
### What would you like to be added?

Kusion accepts spec as an input format in addition to KCL

### Why is this needed?

Compatibility requirement

### How to do it?

Add the `--spec-file` flag to `kusion preview` and `kusion apply`. Kusion will then read file from `--spec-file` as a spec input instead of generating it from KCL.

### PR DESCRIPTION
**Pull request by panshuai-ps**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Kusion accepts spec as an input format in addition to KCL
close #396 

#### Does this PR introduce a user-facing change?
NONE

### CODE REVIEW

File: docs/cmd/zh_CN/kusion.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ kusion [flags]
 ### SEE ALSO
 
 * [kusion apply](kusion_apply.md)	 - 将各类资源的运维意图 apply 至多种运行时里 (Kubernetes、AWS/阿里云以及自建系统等)
+* [kusion check](kusion_check.md)	 - Check if KCL configurations in current directory ok to compile
```
Comment (SparkYuan) [inline docs/cmd/zh_CN/kusion.md:22]: translate to Chinese

File: docs/cmd/zh_CN/kusion_preview.md [hunk 2]
```diff
@@ -46,11 +46,12 @@ kusion preview [flags]
   -o, --output string            指定输出文件
   -O, --overrides strings        指定配置覆盖路径和值
   -Y, --setting strings          指定命令行配置文件
+      --spec-file string         指定 spec 文件作为输入
```
Comment (SparkYuan) [inline docs/cmd/zh_CN/kusion_preview.md:49]: 文件 -> 文件路径

File: go.mod [hunk 3]
```diff
@@ -62,6 +63,14 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require (
```
Comment (SparkYuan) [inline go.mod:66]: please combine this require section into the one below

File: pkg/cmd/preview/options.go [hunk 4]
```diff
@@ -57,6 +59,38 @@ func (o *PreviewOptions) Validate() error {
...
+	}
+	absWD, err := filepath.Abs(o.WorkDir)
+	if err != nil {
+		return err
+	}
+	rel, err := filepath.Rel(absWD, absSF)
+	if err != nil {
+		return err
+	}
+	if rel[:3] == ".."+string(filepath.Separator) {
```
Comment (SparkYuan) [inline pkg/cmd/preview/options.go:91]: This validation is too strict. If a Spec file is specified with an absolute path in the current working directory, would it be considered an error?

File: pkg/cmd/preview/options.go [hunk 5]
```diff
@@ -57,6 +59,38 @@ func (o *PreviewOptions) Validate() error {
...
+		return fmt.Errorf("spec file must be located in workDir's directory or its subdirectory")
```
Comment (SparkYuan) [inline pkg/cmd/preview/options.go:92]: workDir's directory? Is it a typo?

File: pkg/cmd/preview/preview.go [hunk 6]
```diff
@@ -73,4 +73,6 @@ func (o *PreviewOptions) AddPreviewFlags(cmd *cobra.Command) {
 		i18n.T("Ignore differences of target fields"))
 	cmd.Flags().StringVarP(&o.Output, "output", "o", "",
 		i18n.T("Specify the output format"))
+	cmd.Flags().StringVarP(&o.SpecFile, "spec-file", "", "",
```
Comment (SparkYuan) [inline pkg/cmd/preview/preview.go:76]: please add an example for this new flag

File: pkg/cmd/preview/options.go (hunk 4 above)
Comment (panshuai-ps) [reply pkg/cmd/preview/options.go:95]: `filepath.Rel` can calculate the relative path between two path, regardless of whether the input path is a relative path or an absolute path. BTW, the relevant usercase has been added.

Comment (panshuai-ps) [reply pkg/cmd/preview/options.go:95]: `filepath.Rel` can calculate the relative path between two path, regardless of whether the input path is a relative path or an absolute path. BTW, the relevant usercase has been added.

File: pkg/cmd/spec/generator.go [hunk 7]
```diff
@@ -72,3 +77,18 @@ func GenerateSpec(o *generator.Options, project *projectstack.Project, stack *pr
...
+	b, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	decoder := yaml.NewDecoder(bytes.NewBuffer(b))
+	decoder.KnownFields(true)
+	var resources models.Resources
+	if err = decoder.Decode(&resources); err != nil && err != io.EOF {
+		return nil, fmt.Errorf("failed to parse spec file, please check if the file content is valid")
```
Comment (SparkYuan) [inline pkg/cmd/spec/generator.go:91]: "failed to parse spec file" -> "failed to parse the spec file"

**Review by SparkYuan — APPROVED** [review]
lgtm

**Pull request merged**